### ISSUE TITLE
Location parameter in ReactGA.pageview function.

### ISSUE THREAD

**Issue by sergiop** [react-ga/react-ga#210]
Hi all, in my React App I use `ReactGA.initialize` and `ReactGA.pageview` on the first page rendered and `ReactGA.pageview` on every route change.

If I open my react application in a route different from the root, like this one:
```
/myfolder/
```
GA Debug returns me in console:
```
location: http://myreactapp.com/my-folder
page: /my-folder
```
On every following route change, the page value changes correctly its value, but the location doesn't change, like this:

```
location: http://myreactapp.com/my-folder
page: /my-second-route-folder
---
location: http://myreactapp.com/my-folder
page: /my-third-route-folder
---
```
Is this correct? Or I have to update also the location in this way?

```es6
ReactGA.ga('send', {
  hitType: 'pageview',
  location: `${window.location.protocol}//${window.location.host}/`,
  page: window.location.pathname,
});
```

**Comment by SimeonC** [comment]
Hmmm, yea that's a bug. Looking around elsewhere it looks like we should be using:
```
ga('set', 'page', url);
ga('send', 'pageview', url);
```
Whereas currently we are only using the second line.

As a workaround you should be able to use this and we'll get this fixed next version:
```
const page = window.location.pageview;
ReactGA.set({ page });
ReactGA.pageview(page);
```

**Comment by sergiop** [comment]
Mmm... what is `window.location.pageview` @SimeonC?

According to Google Analytics docs: <a href="https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#location">Document location URL (location)</a> is **the full URL (excluding anchor) of the page**, and <a href="https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#page">Document Path (page)</a> is the **The path portion of the page URL**.

This means that if I have an example URL like `http://myreactapp.com/my-folder?param=foo`, I need to send to GA, these data:
```
location: http://myreactapp.com/my-folder?param=foo
page: /my-folder
```

I think I can do it with:
```es6
ReactGA.set({
  location: `
    ${window.location.origin}/
    ${window.location.pathname}
    ${window.location.search}
  `,
});
ReactGA.pageview(window.location.pathname);
```

What do you think @SimeonC @khryst?

**Comment by SimeonC** [comment]
Sorry, typo. Should have been `window.location.pathname` - I've fixed it.
This comes from googles SPA notes here: https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications#tracking_virtual_pageviews

**Comment by sergiop** [comment]
@SimeonC Google notes on SPA doesn't mention the `location` `DL` parameter. So I can assume that I can leave it as is: I need only to update the `page` `DP` value and send the `pageview`. I think we can close this issue: this is not a bug.

I notice that the note also say that, to track a SPA, the best approach is to send two separate commands instead of adding the `page` parameter to `pageview` command.
This way
```
ga('set', 'page', '/new-page.html');
ga('send', 'pageview');
```
Instead of:
```
ga('send', 'pageview', '/new-page.html');
```
According with react-ga sources https://github.com/react-ga/react-ga/blob/master/src/index.js#L189, ReactGA.pageview is using the wrong approach, triggering only the `send` command with the `page` param.
```es6
_gaCommand(trackerNames, 'send', {
  hitType: 'pageview',
  page: path,
  ...extraFields
});
```
Am I wrong?

**Comment by SimeonC** [comment]
That's correct. I was going off the old package I used to use in Angular days here: https://github.com/mgonto/angularytics/blob/master/src/googleHandler.js#L33-L36

I'll leave this open as most people probably expect something more suited to SPA so I'll probably add a `spaPageview(` or something similar highlight the difference. I think more people use React in a SPA so it would make sense to support this like other SPA type packages do.

**Comment by f2net** [comment]
Could I contribute creating this spaPageview(page)?

**Comment by smeijer** [comment]
Has there been made a decision on how to move forward with this issue?

**Comment by RodolfoGS** [comment]
Any solution?
